### ISSUE TITLE
PostgreSQL hfj_search does not reference large objects with `oid`

### ISSUE THREAD

**Issue by tuomoa** [hapifhir/hapi-fhir#2434]
**Describe the bug**

The database schema for `hfj_search` table when using PostgreSQL is:
```
CREATE TABLE public.hfj_search
(
    pid bigint NOT NULL,
    created timestamp without time zone NOT NULL,
    failure_code integer,
    failure_message character varying(500) COLLATE pg_catalog."default",
    last_updated_high timestamp without time zone,
    last_updated_low timestamp without time zone,
    num_found integer NOT NULL,
    preferred_page_size integer,
    resource_id bigint,
    resource_type character varying(200) COLLATE pg_catalog."default",
    search_last_returned timestamp without time zone NOT NULL,
    search_query_string text COLLATE pg_catalog."default",
    search_query_string_hash integer,
    search_type integer NOT NULL,
    search_status character varying(10) COLLATE pg_catalog."default" NOT NULL,
    total_count integer,
    search_uuid character varying(36) COLLATE pg_catalog."default" NOT NULL,
    search_deleted boolean,
    search_param_map oid,
    optlock_version integer,
    CONSTRAINT hfj_search_pkey PRIMARY KEY (pid),
    CONSTRAINT idx_search_uuid UNIQUE (search_uuid)
)
```

The problem is that `search_query_string` column is text, even though it actually references a large object.
`ca/uhn/fhir/jpa/entity/Search.java`:
```
	@Lob()
	@Basic(fetch = FetchType.LAZY)
	@Column(name = "SEARCH_QUERY_STRING", nullable = true, updatable = false, length = MAX_SEARCH_QUERY_STRING)
	private String mySearchQueryString;
```

If we run tools such as [vacuumlo](https://www.postgresql.org/docs/11/vacuumlo.html) it will delete those large objects because there is no references from `oid` type of fields to them and leave the database in inconsistent state (and HAPI FHIR cannot resolve this as it fails to clean up the search).

**Expected behavior**

`hfj_search` table column `search_query_string` type is `oid`

**Environment (please complete the following information):**

 - HAPI FHIR Version 5.2.0

**Comment by tuomoa** [comment]
@jamesagnew any insight on this?

**Comment by tuomoa** [comment]
Fixed by #3016. I'm closing this.
